app/services/personal_key_generator.rb

```diff
@@ -37,7 +37,7 @@ def normalized_code(plaintext_code)
     normed = plaintext_code.gsub(/\W/, '')
     split_length = normed.length / length
     decoded = Base32::Crockford.decode(normed)
-    Base32::Crockford.encode(decoded, length: 16, split: split_length).tr('-', ' ')
```

Comment:
I believe this change will render existing recovery codes invalid.

This line:
```
normed = plaintext_code.gsub(/\W/, '')
```

ignores anything that isn't a `\w` character. So these inputs are treated the same:
```
abcd-1234
abcd?1234
abcd 1234
abcd!!!!!!!!1234
```

Why not continue to have the hashed key use spaces, but presented back via `create` with hyphens?

e.g. this instead
```diff
diff --git a/app/services/personal_key_generator.rb b/app/services/personal_key_generator.rb
index e35f4cf5..e6176b6e 100644
--- a/app/services/personal_key_generator.rb
+++ b/app/services/personal_key_generator.rb
@@ -15,7 +15,7 @@ class PersonalKeyGenerator
     @user_access_key = make_user_access_key(raw_personal_key)
     user.personal_key = hashed_code
     user.save!
-    raw_personal_key
+    raw_personal_key.tr(' ', '-')
   end
 
   def verify(plaintext_code)
```